bqskit/runtime/base.py

```diff
@@ -41,15 +41,22 @@ def __init__(
         conn: Connection,
         total_workers: int,
         process: Process | None = None,
-        num_tasks: int = 0,
     ) -> None:
         """Construct an employee with all resources idle."""
         self.conn: Connection = conn
         self.total_workers = total_workers
         self.process = process
-        self.num_tasks = num_tasks
+        self.num_tasks = 0
         self.num_idle_workers = total_workers
 
+        self.submit_cache: list[tuple[RuntimeAddress, int]] = []
```

Comment:
Should this be a dictionary instead of a list of tuples?

Reply:
Or maybe ordering matters, so an ordered dict?

Reply:
I don't think so. This isn't as much a key-value pair as just a pair of information.